azure-mgmt-resources/src/test/java/com/microsoft/azure/management/resources/DeploymentsTests.java

```diff
@@ -218,4 +221,59 @@ public void canUpdateVirtualNetworkDeployment() throws Exception {
         Assert.assertNotNull(genericVnet);
         resourceClient.genericResources().delete(rgName, "Microsoft.Network", "", "virtualnetworks", "VNet2", "2015-06-15");
     }
+
+    @Test
+    public void canDeployVirtualNetworkWithContext() throws Exception {
+        final String dpName = "dpA" + testId;
+        final String rgName1 = SdkContext.randomResourceName("rg", 9);
+        final String rgName2 = SdkContext.randomResourceName("rg", 9);
+
+        try {
+            String uuid = SdkContext.randomUuid();
+            Context context = new Context.Builder().withCorrelationRequestId(uuid).build();
+
+            // with context
+            Deployment deployment1 = resourceClient.deployments()
+                    .define(dpName)
+                    .withNewResourceGroup(rgName1, Region.US_SOUTH_CENTRAL)
+                    .withTemplateLink(templateUri, contentVersion)
+                    .withParametersLink(parametersUri, contentVersion)
+                    .withMode(DeploymentMode.COMPLETE)
+                    .beginCreate(context);
+
+            // with context
+            Deployment deployment2 = resourceClient.deployments()
+                    .define(dpName)
+                    .withNewResourceGroup(rgName2, Region.US_SOUTH_CENTRAL)
+                    .withTemplateLink(templateUri, contentVersion)
+                    .withParametersLink(parametersUri, contentVersion)
+                    .withMode(DeploymentMode.COMPLETE)
+                    .beginCreate(context);
```

Comment:
Sample code here. `deployment1` and `deployment2` will share the same `x-ms-correlation-requestid`, while `deployment3` will not.